src/App.tsx

```diff
@@ -1,26 +1,137 @@
 /* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { createTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorNotification } from './components/ErrorNotification';
+import { FiltersType } from './types/FiltersType';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { ErrorTypes } from './types/ErrorTypes';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorTodos, setErrorTodos] = useState(ErrorTypes.NoErrors);
+  const [status, setStatus] = useState(FiltersType.All);
+  const [isLoading, setIsLoading] = useState(true);
+  const [deletingTodosIds, setDeletingTodosIds] = useState<number[]>([]);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const handleErrorClose = () => {
+    setErrorTodos(ErrorTypes.NoErrors);
+  };
+
+  const visibleTodos = todos?.filter(todo => {
+    switch (status) {
+      case FiltersType.Completed:
+        return todo.completed;
+      case FiltersType.Active:
+        return !todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  const activeTodosCount = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  const completedTodos = useMemo(
+    () => todos.filter(todo => todo.completed),
+    [todos],
+  );
```

Comment:
Using `useMemo` is useless here, delete it
Read more about when we should use it
If you need some links, please let me know in your chat